beluga/include/beluga/actions/overlay.hpp

```diff
@@ -30,96 +30,90 @@ namespace beluga::actions {
 
 namespace detail {
 
-/// Implementation detail for an overlay range adaptor object.
-struct overlay_base_fn {
-  /// Overload that implements an overlay of a value in a range.
-  /**
-   * \tparam ExecutionPolicy An [execution policy](https://en.cppreference.com/w/cpp/algorithm/execution_policy_tag_t).
-   * \tparam Range An [input range](https://en.cppreference.com/w/cpp/ranges/input_range).
-   * \tparam MaskRange An [input range](https://en.cppreference.com/w/cpp/ranges/input_range).
-   * \param policy The execution policy to use.
-   * \param range An existing range to apply this action to.
-   * \param mask The mask where the values will be overlaid.
-   * \param mask_value The value to be overlaid.
-   */
-  template <
-      class ExecutionPolicy,
-      class Range,
-      class MaskRange,
-      class Mask,
-      std::enable_if_t<std::is_execution_policy_v<std::decay_t<ExecutionPolicy>>, int> = 0,
-      std::enable_if_t<ranges::range<Range>, int> = 0,
-      std::enable_if_t<ranges::range<MaskRange>, int> = 0>
-  constexpr auto operator()(ExecutionPolicy&& policy, Range& range, MaskRange&& mask, Mask&& mask_value) const
-      -> Range& {
+///\cond detail
+
+template <class MaskRange, class Mask, class ExecutionPolicy = std::execution::sequenced_policy>
+struct overlay_closure {
+ public:
+  static_assert(std::is_execution_policy_v<ExecutionPolicy>);
+  static_assert(ranges::range<MaskRange>);
+
+  constexpr overlay_closure(MaskRange mask, Mask mask_value)
+      : policy_{std::execution::seq}, mask_(std::move(mask)), mask_value_(std::move(mask_value)) {}
+
+  constexpr explicit overlay_closure(ExecutionPolicy policy, MaskRange mask, Mask mask_value)
+      : policy_(std::move(policy)), mask_(std::move(mask)), mask_value_(std::move(mask_value)) {}
```

Comment:
nit: Use uniform initialization when possible.
```suggestion
      : policy_{std::move(policy)}, mask_{std::move(mask)}, mask_value_{std::move(mask_value)} {}
```